### PR TITLE
Update Saga README.md and fix some bugs.

### PR DESCRIPTION
**Pull request by MabinGo**
Signed-off-by: MabinGo <bin.ma@huawei.com>

### CODE REVIEW

File: saga-demo/README.md [hunk 1]
```diff
@@ -24,9 +24,9 @@ User ---> Saga -----> flight booking service
 ```
 
 ## Running Demo
-1. run the following command to create docker images in saga project root folder
+1. when Saga service is ready, run the following command in saga/saga-demo/ directory to create microservice docker images.
```
Comment (seanyinx) [inline saga-demo/README.md]: perhaps it's better to ask the user to create docker images in saga root folder with command
`mvn package -DskipTests -Pdocker -Pdemo`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,74 @@
 # Saga [![Build Status](https://travis-ci.org/ServiceComb/saga.svg?branch=master)](https://travis-ci.org/ServiceComb/saga?branch=master) [![Coverage Status](https://coveralls.io/repos/github/ServiceComb/saga/badge.svg?branch=master)](https://coveralls.io/github/ServiceComb/saga?branch=master) [![License](https://img.shields.io/badge/license-Apache%202-4EB1BA.svg)](https://www.apache.org/licenses/LICENSE-2.0.html)
 
+## Purpose
+Saga is implemented based on the compensation transaction driven mechanism, which provides a simple way to help users solve the data consistency problems encountered in micro-service applications.
```
Comment (seanyinx) [inline README.md]: Saga is a type of [Compensating Transaction pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/compensating-transaction)

File: README.md [hunk 3]
```diff
@@ -1,2 +1,74 @@
...
+Reference documentation is available on the [ServiceComb website][servicecomb-website].
+
+[servicecomb-website]: http://servicecomb.io/
+
+## Major Architecture of Saga
+* saga-core(transaction and compensation handled)
+* saga-format(events serialized and deserialized)
+* saga-transports(protocol processing as rest and rpc)
+* saga-discovery(discovery service center)
+* saga-spring(running architecture)
```
Comment (seanyinx) [inline README.md]: * saga-core (transaction and compensation handling logic)
* saga-format (data serialization and deserialization)
* saga-transports (communication protocol implementation such as rest or rpc in the future)
* saga-discovery (service discovery)
* saga-spring (restful service framework)

File: README.md [hunk 4]
```diff
@@ -1,2 +1,74 @@
...
+![Saga](docs/static_files/saga.png) 
+
+## Prerequisites
+You will need:
+1. [Oracle JDK 1.8+][jdk]
+2. [Maven 3.x][maven]
+3. [Docker][docker]
+4. [MySQL][mysql]
+5. [Docker compose(optional)][docker_compose]
+6. [Docker machine(optional)][docker_machine]
```
Comment (seanyinx) [inline README.md]: service center may be needed (optional)

File: docs/api/api.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      }
+    }
+  ]
+}
+```
+JSON parameters:
+- policy - support `BackwardRecovery` or `ForwardRecovery`.
+- requests - transactions array.
+  - id - user-defined service id.
+  - type - support `rest` or `rpc`.
```
Comment (seanyinx) [inline docs/api/api.md]: rest only for now

File: docs/api/api.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        }
+      }
+    }
+  ]
+}
+```
+JSON parameters:
+- policy - support `BackwardRecovery` or `ForwardRecovery`.
+- requests - transactions array.
+  - id - user-defined service id.
```
Comment (seanyinx) [inline docs/api/api.md]: request id instead of service id. it shall be unique among this collection of requests

File: docs/api/api.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+  - type - support `rest` or `rpc`.
+  - serviceName - user-defined service name.
+  - parents - specify user-defined service id. It means this request is only executed after all requests in the parents field are completed.
```
Comment (seanyinx) [inline docs/api/api.md]: parent request ids

File: saga-demo/README.md (hunk 1 above)
Comment (MabinGo) [reply saga-demo/README.md]: modified completed

File: README.md (hunk 2 above)
Comment (MabinGo) [reply README.md]: modified completed

File: README.md (hunk 3 above)
Comment (MabinGo) [reply README.md]: modified completed

File: README.md (hunk 4 above)
Comment (MabinGo) [reply README.md]: modified completed

File: docs/api/api.md (hunk 5 above)
Comment (MabinGo) [reply docs/api/api.md]: modified completed

File: docs/api/api.md (hunk 6 above)
Comment (MabinGo) [reply docs/api/api.md]: modified completed

File: docs/api/api.md (hunk 7 above)
Comment (MabinGo) [reply docs/api/api.md]: modified completed

**Pull request merged**